backend/selectgen.ml

```diff
@@ -888,6 +894,9 @@ method emit_expr (env:environment) exp =
               let rd = self#insert_op_debug env new_op dbg r1 rd in
               set_traps_for_raise env;
               Some rd
+          | Iintofvalue | Ivalueofint ->
+              let rs = self#emit_tuple env simple_args in
+              Some (self#insert_op_debug env new_op dbg rs rs)
```

Comment:
Please remove this special case: it forces arg and result to be the same `Reg.t` (as in the case of `opaque`) instead of using `ty` (i.e., the result of `oper_result_type op`) to create a `Reg.t` of the correct result "type". `Reloadgen` needs to be adjusted too: just remove special case for `Iintofvalue` and `Ivalueofint` for now. It means we may end up using different registers for arg and res (and the assert in Emit will fail), but I want to see a case like this, and I couldn't come up with an example that leads to it.

Here is a test that shows why this is needed:
```
external unsafe_to_value
  :  (nativeint[@unboxed])
  -> 'a
  = "caml_native_pointer_to_value_bytecode" "caml_native_pointer_to_value"
[@@noalloc] [@@no_effects] [@@no_coeffects][@@builtin]

let[@inline never] bar x = (Sys.opaque_identity x) |> ignore

let foo x =
  x
  |> Nativeint.add 7n
  |> unsafe_to_value
  |> fun x -> bar x; x
```
The generated frametable for `foo`, it has no recorded live values for the call to bar before this fix, and has one value after it.  We can confirm using compiler flag `-dlinear` for example that the live set before the call to `bar` after the fix contains "Val", as indicated by the "*":
``` 
{spilled-x/33[s[i:0]]*}
call "camlT__bar_262" R/0[%rax] {t.ml:19,14-19}
```